extension/jaegerremotesampling/config.go

```diff
@@ -15,24 +15,52 @@
 package jaegerremotesampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/jaegerremotesampling"
 
 import (
+	"errors"
+	"time"
+
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+var (
+	errTooManySources = errors.New("too many sources specified, has to be either 'file' or 'remote'")
+	errNoSources      = errors.New("no sources specified, has to be either 'file' or 'remote'")
 )
 
 // Config has the configuration for the extension enabling the health check
 // extension, used to report the health status of the service.
 type Config struct {
 	config.ExtensionSettings      `mapstructure:",squash"`
-	configgrpc.GRPCServerSettings `mapstructure:",squash"`
-	configgrpc.GRPCClientSettings `mapstructure:"remote"`
+	confighttp.HTTPServerSettings `mapstructure:"http"`
```

Comment:
Do you allow enabling one of the http/grpc? How do you know when one is enabled vs the other? Probably you need this to be pointer.

Reply:
Both are enabled by default, but your question made me realize that users might not want to enable both, so, we need a way to either allow users to disable those ports, or only open when explicitly set. I think I prefer the second, so, making it a pointer will indeed be a good solution.